### ISSUE TITLE
Defer parse errors that are inside a `try` until execution

### ISSUE THREAD

**Issue by NotTheDr01ds** [nushell/nushell#13703]
### Related problem

Parse errors in a `try` block cannot be handled by the `try`/`catch` mechanism.

```nu
try {
  use ~/.local/share/nushell/modules/foo
}

# => Result
Error: nu::parser::module_not_found

  × Module not found.
   ╭─[entry #12:2:7]
 1 │ try {
 2 │   use ~/.local/share/nushell/modules/foo
   ·       ─────────────────┬────────────────
   ·                        ╰── module ~/.local/share/nushell/modules/foo not found
 3 │ }
   ╰────
  help: module files and their paths must be available before your script is run as parsing occurs before anything is evaluate
```

### Describe the solution you'd like

XY issue - The Y is that I simply want to be able to suppress parse time errors, especially when loading my startup config and a module isn't available on the system. As discussed in #13210, errors today stop subsequent config from loading.  Sometimes (as in #13210), I want to display those errors and continue running, but other times I know that a module may-or-may-not be available depending on which system the config is running on (or even other conditions).

Is there some way that the parser could 'defer" the error until execution time when it occurs inside a `try` block?

**Comment by sholderbach** [comment]
I feel like your request should be narrowed to the parse errors relating to the availability/validity of a module.

It seems bad suppressing any parse error not down to:
1. `use`/`source` nothing found under the given const
2. (maybe, but inclined to also reject it) `use`/`source` failing to parse the stuff inside.

**Comment by NotTheDr01ds** [comment]
> be narrowed to the parse errors relating to the availability/validity of a module

Agreed - (1) is certainly the main use case, and I can't think of any other type of parse error off the top of my head that should need to be suppressed/handled in a `try`.

**Comment by sbrow** [comment]
I also would like to be able to catch parse errors. (possibly with a flag? `try --all` or `try --include-parse`)

In my case I'm running variable external scripts.

```nu
def run-script-a [
  itemId: string
] {
  let script = (
    $env.TOOL_INSTALLATION_DIR
    | path join scripts script-a.nu
  )

  ^$script --force --json $itemId | from json
}
```

```bash
Error: nu::parser::unknown_flag

  × The `script-a.nu` command doesn't have flag `force`.
   ╭─[<commandline>:1:6]
 1 │ main --force -u 3 15
   ·      ───┬───
   ·         ╰── unknown flag
   ╰────
  help: Available flags: --user-limit(-u), --help(-h). Use `--help` for more information.
